src/lqfit.jl

```diff
@@ -0,0 +1,334 @@
+
+    # regenerate histograms
+    peakhists = [fit(Histogram, lq_compton_bands[i], min_lq[i]:bin_width[i]:max_lq[i]) for i in eachindex(lq_compton_bands)]
+
+    # reestimate peak parameters
+    peakstats = StructArray(estimate_single_peak_stats_psd.(peakhists))
+
+    (
+        ;
+        peakhists,
+        peakstats,
+        min_lq,
+        e_unit,
+        max_lq,
+        mean_peak_pos,
+        std_peak_pos,
+        simple_pars_lq_μ,
+        simple_pars_error_lq_μ,
+        mean_peak_sigma,
+        std_peak_sigma,
+        simple_pars_lq_σ,
+        simple_pars_error_lq_σ
+    )
+end
+export generate_lq_compton_bands
+
+
+
+"""
+    fit_lq_compton(peakhists::Array, peakstats::StructArray, compton_bands::Array{T}) where T<:Real
+
+Fit the A/E Compton bands using the `f_lq_compton` function consisting of a gaussian SSE peak and a step like background for MSE events.
+
+# Returns
+    * `result`: Dict of NamedTuples of the fit results containing values and errors for each compton band
+    * `report`: Dict of NamedTuples of the fit report which can be plotted for each compton band
+"""
+function fit_lq_compton(peakhists::Vector{<:Histogram}, peakstats::StructArray, compton_bands::Array{T},; pars_lq::NamedTuple{(:μ, :μ_err, :σ, :σ_err)}=NamedTuple{(:μ, :μ_err, :σ, :σ_err)}(nothing, nothing, nothing, nothing), uncertainty::Bool=false) where T<:Unitful.Energy{<:Real}
+    # create return and result dicts
+    result = Dict{T, NamedTuple}()
+    report = Dict{T, NamedTuple}()
+    # iterate throuh all peaks
+    for (i, band) in enumerate(compton_bands)
+        # get histogram and peakstats
+        h  = peakhists[i]
+        ps = peakstats[i]
+        if !isnothing(pars_lq.μ)
+            ps = merge(peakstats[i], (μ = f_lq_μ(band, pars_lq.μ), σ = f_lq_σ(band, pars_lq.σ)))
+        end
+        # fit peak
+        result_band, report_band = nothing, nothing
+        try
+            result_band, report_band = fit_single_lq_compton(h, ps, ; uncertainty=uncertainty)
+        catch e
+            @warn "Error fitting band $band: $e"
+            continue
+        end
+        # save results
+        result[band] = result_band
+        report[band] = report_band
+    end
+    return result, report
+end
+export fit_lq_compton
+
+
+"""
+    fit_single_lq_compton(h::Histogram, ps::NamedTuple{(:peak_pos, :peak_fwhm, :peak_sigma, :peak_counts, :mean_background, :μ, :σ), NTuple{7, T}}; uncertainty::Bool=true) where T<:Real
+
+Perform a fit of the peakshape to the data in `h` using the initial values in `ps` while using the `f_lq_compton` function consisting of a gaussian SSE peak and a step like background for MSE events.
+
+# Returns
+    * `result`: NamedTuple of the fit results containing values and errors
+    * `report`: NamedTuple of the fit report which can be plotted
+"""
+function fit_single_lq_compton(h::Histogram, ps::NamedTuple; uncertainty::Bool=true)
```

Comment:
Did you play with the `priors` or are these also the same ones? Maybe also call the `A/E` routines here.